lms/djangoapps/courseware/url_helpers.py

```diff
@@ -50,15 +53,15 @@ def get_redirect_url(course_key, usage_key):
     return redirect_url
 
 
-def get_redirect_url_for_global_staff(course_key, _next):
+def get_redirect_url_for_global_staff(course_key, _next=None):
     """
     Returns the redirect url for staff enrollment
 
     Args:
         course_key(str): Course key string
-        _next(str): Redirect url of course component
+        _next(str): Redirect URL of course component (defaults to the current URL)
     """
     redirect_url = ("{url}?next={redirect}".format(
         url=reverse('enroll_staff', args=[unicode(course_key)]),
-        redirect=_next))
+        redirect=_next or RequestCache.get_current_request().path))
```

Comment:
I thought I understood this but trying to reply to your comment shows that I don't. :-) I should have removed this method rather than fixing it, because the code no longer redirects global staff in a special way. I'll remove it.